crates/composer/src/main.rs

```diff
@@ -15,23 +15,28 @@ fn main() -> Result<()> {
     let (_stream, stream_handle) = OutputStream::try_default()?;
 
     loop {
-        if let Err(err) = {
-            // Should be more than enough.
-            let mut buf = [0; 1000];
-            let (number_of_bytes, _) = socket.recv_from(&mut buf)?;
-
-            let event: Event = bincode::deserialize(&buf)?;
-            println!("Received an event ({number_of_bytes} bytes): {:?}", event);
-
-            // FIXME: do the decoding and file reading outside the loop
-            let file = BufReader::new(File::open("src/sound_samples/click.wav")?);
-            let source = Decoder::new(file)?;
-            stream_handle.play_raw(source.convert_samples())?;
-
-            Ok::<(), eyre::Error>(())
-        } {
-            eprintln!("Could not process data-gram: {:?}", err);
+        if let Err(err) = handle_datagram(&socket, &stream_handle) {
+            eprintln!(
+                "Could not process datagram. Ignoring and continuing. {:?}",
+                err
+            );
             continue;
         }
     }
 }
+
+fn handle_datagram(socket: &UdpSocket, output_stream: &OutputStreamHandle) -> Result<()> {
+    // Size up to max normal network packet size
+    let mut buf = [0; 1500];
+    let (number_of_bytes, _) = socket.recv_from(&mut buf)?;
+
+    let event: Event = bincode::deserialize(&buf)?;
```

Comment:
The current code works, but it has 2 assumptions that currently hold, but are unnecessary
- We use a deserializer (bincode) that copes with extra trailing data (zeroes)
- We zero the buffer for every call

It is simple to remove these assumptions, and match the canonical usage example from `recv_from()`, so I suggest doing that: 
```suggestion
    let mut buf = [0; 1500];
    let (number_of_bytes, _) = socket.recv_from(&mut buf)?;

    let event: Event = bincode::deserialize(&buf[..number_of_bytes])?;
```

Reply:
Dang, I'm not sure why this didn't appear to me. Done 25fbe2f61be63b55bfb0e545f650bb5feeff130a

Thanks!